src/server.py

```diff
@@ -0,0 +1,42 @@
+import socket
+import datetime
+
+sock = socket.socket(
+    socket.AF_INET,
+    socket.SOCK_STREAM,
+    socket.IPPROTO_TCP
+)
+
+address = ('127.0.0.1', 3000)
+
+sock.bind(address)
+
+sock.listen(1)
+
+print('--- Starting server on port 3000 at {} ---'.format(datetime.datetime.today()))
+
+buffer_lenth = 2048
+
+message_complete = False
+
+try:
+    conn, addr = sock.accept()
+
+    while not message_complete:
+        part = conn.recv(buffer_lenth)
+        message = part.decode('utf8')
+        if not part:
```

Comment:
This will never happen. The `.recv()` method waits until it gets any data, so if it does not receive anything the server will just hang. A better check would be on the length of `part`, but that will also hang if the received data is a multiple of the buffer length.